saleor/graphql/checkout/types.py

```diff
@@ -973,7 +976,23 @@ def calculate_shipping_price(data):
 
     @staticmethod
     def resolve_lines(root: models.Checkout, info: ResolveInfo):
-        return CheckoutLinesByCheckoutTokenLoader(info.context).load(root.token)
+        @allow_writer_in_context(info.context)
+        def create_discounts_and_fetch_lines(lines_info):
+            CheckoutLinesByCheckoutTokenLoader(info.context).clear(root.token)
+            create_discount_objects_for_order_promotions(
```

Comment:
IMHO I don't think that this is a proper approach:
- by calling `.clear` we remove the details from DD. It means that we will fetch lines once again
- previously `create_discount_objects_for_order_promotions` was called when prices were expired, not always - this could produce undefined behaviour or at least increased amount of DB queries.